### ISSUE TITLE
[Arch] Sidecar estimators lack a shared output contract / Protocol — lock the envelope before Phase-2/3 estimators

### ISSUE THREAD

**Issue by magedzamzam** [magedzamzam/beacon-trader#70]
## Summary

The shadow-analytics estimators (`analysis/estimators.py`, harnessed by `analysis/sidecar.py`) have **no shared output contract** — each is a bare `fn(ctx) -> Optional[dict]` returning an ad-hoc shape. This contrasts sharply with the TA side (`ta/registry.py`), which is a clean spec-based registry. Locking a minimal estimator `Protocol` + a declared output envelope **now**, while there are only ~4 estimators, avoids a costly retrofit once Phase-2/3 estimators (#54 HMM, #55 XGBoost/LightGBM, #56 microstructure) land under epic #51. Non-breaking: sidecar output is shadow data only, never gates execution.

## Current state

- Estimators collected into a plain dict `ESTIMATORS` (`estimators.py:330`), registered via `sidecar.register_estimator` (`sidecar.py:43`). Only a `Callable` type alias exists (`sidecar.py:39`) — **no base class / `Protocol`**.
- Sync-vs-async disambiguated at call time via `inspect.isawaitable` (`sidecar.py:109`).
- **Return shapes are inconsistent per estimator:**
  - `regime` → `{label, adx, hurst, rvol, ...}`
  - `kalman` → `{level, slope, method}`
  - `knn` → `{k, win_rate, neighbours}`
  - `structure_magnet` → `{map_version_id, per_tf, ...}`
- A common shape is **declared but unused**: `structure.py:250 feature_contribution(name, value, direction, weight, confidence)` documents *"the shared shape that structure/magnets emit — identical to what regime/TA/bayes will emit,"* but no estimator actually returns it. The contract is aspirational and unenforced.

## Why it matters

`analysis/feature_vector.py` and `analysis/bayes.py` already have to special-case each estimator's shape when flattening into the model feature vector. Every new estimator adds another bespoke shape and another special-case. A declared envelope lets the feature-vector assembler consume estimators uniformly.

## Proposed fix (minimal, non-breaking)

1. Define an `Estimator` `Protocol` in `sidecar.py` (or a small `base.py`), e.g.:
   ```python
   class Estimator(Protocol):
       name: str
       async def __call__(self, ctx: AnalyticsCtx) -> "EstimatorResult | None": ...
   ```
2. Define a light `EstimatorResult` envelope: `{value, label?, contributions?: list[FeatureContribution], detail?: dict}` — realize the intent already written in `feature_contribution()`.
3. Migrate the 4 existing estimators to return the envelope (keep the rich `detail` sub-dict for back-compat with what's persisted today).
4. Have `feature_vector.assemble` / `bayes` read the envelope uniformly instead of per-estimator branches.

## Guardrails
- Shadow-only; **must not** gate or alter execution.
- Keep the persisted `SignalAnalytics.analytics` JSON back-compatible (envelope can carry the old dict under `detail`).
- Suite stays green (`123 passed`).

Related: epic #51; #53 (Phase 1 estimators); #54/#55/#56 (future phases that will benefit most).

_Filed by the scheduled architecture review (2026-07-13). Pillar 1 (interface abstraction). Report: `beacon-claude-ai/trading-bot-tech/reports/arch-review_20260713.md`._

**Comment by magedzamzam** [comment]
Done in 6aeae93 (pushed to main).

Locked the estimator output envelope before the Phase-2/3 estimators land. New **`analysis/contract.py`** defines:
- **`Estimator`** — a `runtime_checkable` Protocol for the estimator call contract (callable of ctx → JSON-able dict, sync or async)
- **`EstimatorResult`** — the result type
- **`feature_contribution()`** — the canonical `(name, value, direction, weight, confidence)` shape; `structure.py` now **re-exports** it, so there's a single definition instead of two
- **`estimator_contributions(name, detail)`** — the **single** place that maps each estimator's detail dict to the common contribution shape, replacing the scattered per-estimator branching the issue flagged

**Deliberately NON-BREAKING:** estimators keep returning their rich detail dict (persisted unchanged in `SignalAnalytics.analytics`), so `#62` feature_vector, the internal k-NN, and the `#53`/`#61` reports are untouched. The envelope is realized at the harness layer — `run_estimators` attaches an additive `_contributions` list that a future unified consumer (feature_vector/bayes) can read uniformly without per-estimator special-casing. Migrating each estimator's *return type* to the envelope is deferred precisely because k-NN consumes the raw detail dicts internally; that's a larger change to sequence with the Phase-2 estimator work.

New pure test `test_estimator_contract.py` (7 cases): Protocol match, single-definition re-export, contribution shape for every estimator, defensive mapping, and harness-attaches-without-mutating-detail. **Full pure suite: 135 passed, 0 failed.**

**Deploy** (core change — baked into every python image):
```
cd ~/beacon-trader && git pull
docker compose build api executor monitor telegram
docker compose up -d api executor monitor telegram
```
